src/design-system/theme/palette.ts

```diff
@@ -141,6 +145,36 @@ const text: Partial<TypeText> = {
 };
 
 const divider = grey["20"];
+
+type ExtendedPaletteColorOptions = PaletteColorOptions & {
+  [key: string]: string | any;
+};
+
+declare module "@mui/material/styles/createPalette" {
+  export interface PaletteOptions {
+    custom?: ExtendedPaletteColorOptions;
+  }
+}
+
+const custom: ExtendedPaletteColorOptions & Record<string, any> = {
+  backgroundPrimary: "#FFFFFF",
+  backgroundSecondary: "#FCFCFD",
+  backgroundTertiary: "#F9FAFB",
+  backgroundQuaternary: "#F2F4F7",
+  backgroundContrast: "#292929",
+  textPrimary: "#1C1C1C",
+  textSecondary: "#4F4F4F",
+  textTertiary: "#696969",
+  textMuted: "#9C9C9C",
+  textDisabled: "#B5B5B5",
+  textLink: "#4F00BA",
+  textContrast: "#FFFFFF",
+  borderPrimary: "#E0E0E0",
+  borderSecondary: "#DEDEDE",
+  borderTertiary: "#D6D6D6",
+  borderQuaternary: "#CFCFCF",
+};
```

Comment:
Does this work?

```suggestion
const custom = {
  backgroundPrimary: "#FFFFFF",
  backgroundSecondary: "#FCFCFD",
  backgroundTertiary: "#F9FAFB",
  backgroundQuaternary: "#F2F4F7",
  backgroundContrast: "#292929",
  textPrimary: "#1C1C1C",
  textSecondary: "#4F4F4F",
  textTertiary: "#696969",
  textMuted: "#9C9C9C",
  textDisabled: "#B5B5B5",
  textLink: "#4F00BA",
  textContrast: "#FFFFFF",
  borderPrimary: "#E0E0E0",
  borderSecondary: "#DEDEDE",
  borderTertiary: "#D6D6D6",
  borderQuaternary: "#CFCFCF",
};

declare module "@mui/material/styles/createPalette" {
  export interface PaletteOptions {
    custom?: typeof custom;
  }
}
```

You may also need to say (will need to import `MuiPaletteOptions`:
```js
export interface PaletteOptions implements MuiPaletteOptions {
```